src/main/kotlin/com/mparticle/kits/GoogleAnalyticsFirebaseGA4Kit.kt

```diff
@@ -464,16 +477,39 @@ class GoogleAnalyticsFirebaseGA4Kit : KitIntegration(), KitIntegration.EventList
         }
 
     }
+    private fun hashMapToBundle(hashMap: HashMap<String, Any>): PickyBundle {
+        val bundle = PickyBundle()
+        for ((key, value) in hashMap) {
+            when (value) {
+                is String -> {
+                    bundle.putString(key, value)
+                }
 
-    private fun getBundle(product: Product): PickyBundle {
-        return PickyBundle()
-            .putLong(FirebaseAnalytics.Param.QUANTITY, product.quantity.toLong())
-            .putString(FirebaseAnalytics.Param.ITEM_ID, product.sku)
-            .putString(FirebaseAnalytics.Param.ITEM_NAME, product.name)
-            .putString(FirebaseAnalytics.Param.ITEM_CATEGORY, product.category)
-            .putDouble(FirebaseAnalytics.Param.PRICE, product.unitPrice)
+                is Int -> {
+                    bundle.putInt(key, value)
+                }
+
+                is Double -> {
+                    bundle.putDouble(key, value)
+                }
+
+                is Long -> {
+                    bundle.putLong(key, value)
+                }
+            }
```

Comment:
It won't let me make a code suggestion because there's a deleted line in the middle, but this can be significantly condensed like this:

```
when (value) {
    is String -> bundle.putString(key, value)
    is Int -> bundle.putInt(key, value)
    is Double -> bundle.putDouble(key, value)
    is Long -> bundle.putLong(key, value)
}
```